packages/puppeteer-core/src/node/PuppeteerNode.ts

```diff
@@ -150,41 +150,67 @@ export class PuppeteerNode extends Puppeteer {
    * @param options - Options to configure launching behavior.
    */
   launch(options: LaunchOptions = {}): Promise<Browser> {
-    const {browser: product = this.defaultBrowser} = options;
-    this.#lastLaunchedBrowser = product;
+    const {browser = this.defaultBrowser} = options;
+    this.#lastLaunchedBrowser = browser;
+    switch (browser) {
+      case 'chrome':
+        this.defaultBrowserRevision = PUPPETEER_REVISIONS.chrome;
+        break;
+      case 'firefox':
+        this.defaultBrowserRevision = PUPPETEER_REVISIONS.firefox;
+        break;
+      default:
+        throw new Error(`Unknown product: ${browser}`);
+    }
+    this.#launcher = this.#getLauncher(browser);
     return this.#launcher.launch(options);
   }
 
   /**
    * @internal
    */
-  get #launcher(): BrowserLauncher {
-    if (
-      this.#_launcher &&
-      this.#_launcher.browser === this.lastLaunchedBrowser
-    ) {
-      return this.#_launcher;
+  #getLauncher(browser: SupportedBrowser): BrowserLauncher {
+    if (this.#launcher && this.#launcher.browser === browser) {
+      return this.#launcher;
     }
-    switch (this.lastLaunchedBrowser) {
+    switch (browser) {
       case 'chrome':
-        this.defaultBrowserRevision = PUPPETEER_REVISIONS.chrome;
-        this.#_launcher = new ChromeLauncher(this);
-        break;
+        return new ChromeLauncher(this);
       case 'firefox':
-        this.defaultBrowserRevision = PUPPETEER_REVISIONS.firefox;
-        this.#_launcher = new FirefoxLauncher(this);
-        break;
+        return new FirefoxLauncher(this);
       default:
-        throw new Error(`Unknown product: ${this.#lastLaunchedBrowser}`);
+        throw new Error(`Unknown product: ${browser}`);
     }
-    return this.#_launcher;
   }
 
+  /**
+   * The default executable path for a given ChromeReleaseChannel.
+   */
+  executablePath(channel: ChromeReleaseChannel): string;
+  /**
+   * The default executable path given LaunchOptions.
+   */
+  executablePath(options: LaunchOptions): string;
   /**
    * The default executable path.
    */
-  executablePath(channel?: ChromeReleaseChannel): string {
-    return this.#launcher.executablePath(channel);
+  executablePath(): string;
+  executablePath(optsOrChannel?: ChromeReleaseChannel | LaunchOptions): string {
+    if (optsOrChannel === undefined) {
+      return this.#getLauncher(this.lastLaunchedBrowser).executablePath(
+        undefined,
+        /* validatePath= */ false,
```

Comment:
am I right to say that the previous behaviour was to always validate the path in `executablePath`, so this could be a breaking change? since it no longer checks if the path exists from what I can see.

is there any way to still make it validate the path or would that have to be manually done now

Reply:
right, it could be viewed as a breaking change but the documented behavior for this method always been that it returns the default path for the browser (without any validation) and the validation of the path was probably done for the cases when it is called to launch the browser internally. Note that the path is still validated when launching the browser.